src/components/rangeslider/defaults.js

```diff
@@ -12,14 +12,13 @@ var Lib = require('../../lib');
 var attributes = require('./attributes');
 
 
-module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
+module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, axName, counterAxes) {
 
-    if(!layoutIn.xaxis || !layoutOut.xaxis) return;
+    if(!layoutIn[axName].rangeslider) return;
```

Comment:
I did this so that we could say `rangeslider: true` and be done with it - no need to use an object. If that's undesirable we can change it.